app/org/sagebionetworks/bridge/services/StudyService.java

```diff
@@ -519,6 +520,11 @@ private void checkViolationConstraints(Study study) {
                         throwConstraintViolation(plan, study);
                     }
                 }
+                for (String eventId : Schedule.EVENT_ID_SPLITTER.split(schedule.getEventId())) {
+                    if (!eventKeys.contains(eventId)){
+                        throwConstraintViolation(plan, study);
+                    }
+                }
```

Comment:
Also add validation that these event strings don't have colons in the StudyValidator